src/libstore-c/nix_api_store.cc

```diff
@@ -216,4 +216,94 @@ nix_err nix_store_copy_closure(nix_c_context * context, Store * srcStore, Store
     NIXC_CATCH_ERRS
 }
 
+nix_err nix_store_drv_from_path(
+    nix_c_context * context,
+    Store * store,
+    const StorePath * path,
+    void (*callback)(void * userdata, const nix_derivation * drv),
+    void * userdata)
+{
+    if (context)
+        context->last_err_code = NIX_OK;
+    try {
+        nix::Derivation drv = store->ptr->derivationFromPath(path->path);
+        if (callback) {
+            const nix_derivation tmp{drv};
+            callback(userdata, &tmp);
+        }
+    }
+    NIXC_CATCH_ERRS
+}
+
+nix_err nix_derivation_get_outputs(
+    nix_c_context * context,
+    const nix_derivation * drv,
+    void (*callback)(void * userdata, const char * name, const nix_derivation_output * drv_output),
+    void * userdata)
+{
+    if (context)
+        context->last_err_code = NIX_OK;
+    try {
+        if (callback) {
+            for (const auto & [name, result] : drv->drv.outputs) {
+                const nix_derivation_output tmp{result};
+                callback(userdata, name.c_str(), &tmp);
+            }
+        }
+    }
+    NIXC_CATCH_ERRS
+}
+
+nix_err nix_derivation_get_outputs_and_optpaths(
+    nix_c_context * context,
+    const nix_derivation * drv,
+    const Store * store,
+    void (*callback)(void * userdata, const char * name, const nix_derivation_output * drv_output, const StorePath * path),
+    void * userdata)
+{
+    if (context)
+        context->last_err_code = NIX_OK;
+    try {
+        auto value = drv->drv.outputsAndOptPaths(store->ptr->config);
+        if (callback) {
+            for (const auto & [name, result] : value) {
+                const nix_derivation_output tmp_output{result.first};
+
+                if (auto store_path = result.second) {
+                    const StorePath tmp_path{*store_path};
+                    callback(userdata, name.c_str(), &tmp_output, &tmp_path);
+                } else {
+                    callback(userdata, name.c_str(), &tmp_output, nullptr);
+                }
+            }
+        }
+    }
+    NIXC_CATCH_ERRS
+}
+
+const char * nix_derivation_get_structured_attrs(
```

Comment:
This one is easy to do from the JSON

Reply:
How? This gets you the JSON.

Reply:
I mean from the overall derivation JSON. You just take the `structuredAttrs` field of that, if it exists.

Reply:
How do you get the overall derivation JSON?